ee/server/service/service_teams.go

```diff
@@ -245,3 +243,29 @@ func (svc *Service) TeamEnrollSecrets(ctx context.Context, teamID uint) ([]*flee
 
 	return svc.ds.TeamEnrollSecrets(ctx, teamID)
 }
+
+func (svc *Service) ModifyTeamEnrollSecrets(ctx context.Context, teamID uint, secrets []fleet.EnrollSecret) ([]*fleet.EnrollSecret, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.EnrollSecret{TeamID: ptr.Uint(teamID)}, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
+	if secrets == nil {
+		return nil, fleet.NewInvalidArgumentError("secrets", "missing required argument")
+	}
+	team, err := svc.ds.Team(ctx, teamID)
```

Comment:
This is not needed, because you already have the team ID, you have checked that the user can write to this team's enrollment secret (so the team exists). And the table `enrollment_secrets` has a foreign key for `team_id` so if it didn't exist it would fail there.